src/particles/pusher/FieldGather.H

```diff
@@ -211,4 +211,238 @@ void doGatherShapeN (const amrex::ParticleReal xp,
     }
 }
 
+/**
+ * \brief Field gather for a single particle
+ *
+ * \tparam depos_order_xy Order of the transverse shape factor for the field gather
+ * \tparam depos_order_z Order of the longitudinal shape factor for the field gather
+ * \param[in] xp Particle position x
+ * \param[in] yp Particle position y
+ * \param[in] zp Particle position z
+ * \param[in] Aabssqp |a|^2 field at particle position
+ * \param[in] AabssqDxp d/dx |a|^2 field at particle position
+ * \param[in] AabssqDyp d/dy |a|^2 field at particle position
+ * \param[in] a_arr a field array
+ * \param[in] dx 3D cell spacing
+ * \param[in] x_pos_offset offset for converting positions to indexes
+ * \param[in] y_pos_offset offset for converting positions to indexes
+ * \param[in] z_pos_offset offset for converting positions to indexes
+ */
+template <int depos_order_xy, int depos_order_z>
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void doLaserGatherShapeN (const amrex::ParticleReal xp,
+                          const amrex::ParticleReal yp,
+                          const amrex::ParticleReal zp,
```

Comment:
These can probably be passed by const reference. 
```suggestion
void doLaserGatherShapeN (const amrex::ParticleReal& xp,
                          const amrex::ParticleReal& yp,
                          const amrex::ParticleReal& zp,
```